src/central_limit_theorem/SampleRangeSlider.jsx

```diff
@@ -3,95 +3,97 @@ import PropTypes from 'prop-types';
 import { forceNumber } from '../utils';
 import { Slider } from '../LabeledSlider';
 
-export default class SampleRangeSlider extends React.Component {
-    handleFocus(e) {
+export const SampleRangeSlider = ({numberOfSamples,
+    sampleMeansIdx, handleSampleMeansIdx, sampleSize, activeSampleMean,
+    observationIdx, observationData, handleObservationIdx}) => {
+    const handleFocus = (e) => {
         e.target.select();
-    }
-    render() {
-        const handleSampleMeans = (e) => {
-            e.preventDefault();
-            this.props.handleSampleMeansIdx(forceNumber(e.target.value));
-        };
-        const handleObsIdx = (e) => {
```

Comment:
@nbuonin It's your decision because it's kinda your project, but the reason I don't like this syntax is because it's reminiscent of 2012-era JavaScript. Like, yeah, you can define nested functions on one giant function because JavaScript will let you do that. But JavaScript finally has a real class syntax: why not use it? Especially since you're already using babel.

Reply:
Those are good points, and I wasn't really aware of the context. The main reasons I like it are that all the props are in the signature, so its easy to see what's passed in, right where you'd expect to see them.  I also like that params don't have to be prefixed with `this.props` - I think it makes it a bit easier to read, especially with limiting lines to 80 chars.  TBH though, I haven't looked to see if I could do the same with class syntax.

I don't remember where I saw it, either React's docs or in a Lynda tutorial, but the suggestion they made was that if a component doesn't need to maintain its own state, use a function rather than a class.  They argued that it uses less space after its transpiled, and that it uses less memory.  I think those points are really marginal in this particular case though.

Anyway, I think as we do more JS/React/Vue projects, it might be worth agreeing on a common style to facilitate easier reviews, and like in this case, help out when needed.